### ISSUE TITLE
Is it normal to take 20mins on T4?

### ISSUE THREAD

**Issue by cantonalex** [tencent-ailab/V-Express#9]
on a 4 second video?

**Comment by tiankuan93** [comment]
> on a 4 second video?

It is normal! 

Now inference does take a long time. You can try using smaller sampling steps with [`num_inference_steps`](https://github.com/tencent-ailab/V-Express/blob/77e362cd3309b93cacf36a472041033d2f892763/inference.py#L55), which will reduce the time linearly. We recommend using 20-30 steps.

**Comment by cantonalex** [comment]
Thanks one other question @tiankuan93 to do add new audio/lips to existing video is this the correct args 
`python scripts/extract_kps_sequence_and_audio.py \
    --video_path "./destinationVideo.mp4 \
    --kps_sequence_save_path "./kpsOfDestinationVideo.pth" \
    --audio_save_path "./audioToReplaceLips.mp3"`

**Comment by tiankuan93** [comment]
> Thanks one other question @tiankuan93 to add new audio/lips to existing video is this the correct args
> 
> python scripts/extract_kps_sequence_and_audio.py --video_path "./destinationVideo.mp4 --kps_sequence_save_path "./kpsOfDestinationVideo.pth" --audio_save_path "./audioToReplaceLips.mp3"
> 
> like the first video on this comment [#6 (comment)](https://github.com/tencent-ailab/V-Express/issues/6#issue-2320395941)

You're right. Then you can use .mp3 and .pth as inputs for video generation. Note that you need to use the `--retarget_strategy "naive_retarget"` parameter when generating. If the result is not satisfactory, you need to consider choosing a video that is closer to the reference image pose as the target video.

**Comment by cantonalex** [comment]
is a reference image always required? you can't just take a video and say apply audio to this video?

**Comment by tiankuan93** [comment]
> is a reference image always required? you can't just take a video and say apply audio to this video?

1. Currently we need a more frontal image as a reference for generating a video. 
2. The scenario you describe is also possible with a combination of audio, reference image, and target pose. Maybe we can implement some scripts to make the required parameters simpler.

**Comment by cantonalex** [comment]
i'm just a bit confused how @zhanghongyong123456 did this first video? https://github.com/tencent-ailab/V-Express/issues/6#issue-2320395941

im assuming he used a reference image from the same target video but he mentions no retarget strategy

**Comment by tiankuan93** [comment]
I didn't quite get what you meant. If you want an image to talk, but only have an audio(.mp3), then you can use the following script.

```shell
python inference.py \
    --reference_image_path "./test_samples/A.jpg" \
    --audio_path "./test_samples/aud.mp3" \
    --output_path "./output/short_case/A_fix_face_with_aud.mp4" \
    --retarget_strategy "fix_face" \
    --reference_attention_weight 0.95 \
    --audio_attention_weight 3.0
```

**Comment by cantonalex** [comment]
no I purely want to make an existing video talk with new audio without the crop in the export.

There doesn't seem to be a natural lip replacement strategy for an existing video.

**Comment by jasonisme123** [comment]
> is a reference image always required? you can't just take a video and say apply audio to this video?

 maybe you need wav2lip?

**Comment by cantonalex** [comment]
wav2lip quality is lessor than that of this project

**Comment by faraday** [comment]
@cantonalex I couldn't get what you mean in this issue either.
You want to provide the video (your **source** video).
Normally, extract kps sequence script does this:
- with ffmpeg, get the audio of your source video written into audio_save_path
- with InsightFace face analysis app , get faces from frames of the video.
- writes kps sequence.

Now, the source video is only for extracting the kps sequence and get the audio from it (for practical reasons).
You give the reference image and it talks as in the video (hopefully).

What you want is:
- I don't want to provide a reference image (OK)
- then, you could basically take any frame from the source video as the reference image.
- however, I think you don't want this, so do you mean to increase quality by taking multiple reference frames? as in EbSynth or recent AnimateDiff workflows?

**Comment by cantonalex** [comment]
@faraday simply put. I only want to change the lips on a video. So input and output video are the same the only difference is the lip movements.
